pkg/injection/namespace/mapper/dynakubes.go

```diff
@@ -59,7 +61,20 @@ func (dm DynakubeMapper) UnmapFromDynaKube(namespaces []corev1.Namespace) error
 		}
 	}
 
-	return nil
+	return dm.cleanup(namespaces)
+}
+func (dm DynakubeMapper) cleanup(namespaces []corev1.Namespace) error {
+	nsList := make([]string, 0, len(namespaces))
```

Comment:
would add a check that if `len(namespaces) == 0` => _do nothing_

Reply:
probably could do it in `UnmapFromDynaKube`

it wasn't needed before because the `for ... ` did nothing if `namespaces` were empty

Reply:
Good idea! I also added some nice refactorings also based on your input ;) 
https://github.com/Dynatrace/dynatrace-operator/pull/4047/commits/e6ec109d74338fa9c8862575457e6b864ffa3f9c